src/main/resources/assets/js/app/wizard/page/LiveFormPanel.ts

```diff
@@ -698,6 +698,11 @@ export class LiveFormPanel
         this.liveEditPageProxy.onComponentDetached((event: ComponentDetachedFromFragmentEvent) => {
             api.notify.showSuccess(i18n('notify.component.detached', event.getComponentView().getName()));
 
+            const componentPath = event.getComponentView().getComponentPath();
+            const canMarkContentAsReady = this.canMarkContentAsReady(componentPath);
+            if (canMarkContentAsReady) {
+                this.contentWizardPanel.setIsMarkedAsReady(true);
+            }
             this.saveAndReloadOnlyComponent(event.getComponentView());
         });
 
```

Comment:
Let's extract the duplicated code (marked) into a separate method

![image](https://user-images.githubusercontent.com/11612490/63853042-0191f500-c99a-11e9-991e-cc01d8884a3d.png)